polycommit/src/lib.rs

```diff
@@ -672,11 +673,23 @@ pub mod tests {
                 let degree = rand::distributions::Uniform::from(1..=supported_degree).sample(rng);
                 let poly = Polynomial::rand(degree, rng);
 
+                let supported_degree_bounds_after_trimmed = supported_degree_bounds
+                    .iter()
+                    .copied()
+                    .filter(|x| *x >= degree && *x < supported_degree)
+                    .collect::<Vec<usize>>();
+
                 let degree_bound = if let Some(degree_bounds) = &mut degree_bounds {
-                    let range = rand::distributions::Uniform::from(degree..=supported_degree);
-                    let degree_bound = range.sample(rng);
-                    degree_bounds.push(degree_bound);
-                    Some(degree_bound)
+                    if supported_degree_bounds_after_trimmed.len() > 0 && rng.gen() {
```

Comment:
```suggestion
                    if !supported_degree_bounds_after_trimmed.is_empty() > 0 && rng.gen() {
```
otherwise `clippy` might complain